tests/queries/models.py

```diff
@@ -691,6 +691,19 @@ class Ticket23605C(models.Model):
     field_c0 = models.FloatField()
 
 
+class Ingredient(models.Model):
```

Comment:
Is it possible to reuse existing models and develop a more minimal test? (I made the same comment in https://github.com/django/django/pull/8631). I think checking query results is better than examining things like `get_source_expressions()` as that patch does, but hopefully you get the idea. It seems the first failure in the new tests isn't until line 3740. With a bunch of assertions that are already passing, it makes it difficult to tell what this is fixing.